### ISSUE TITLE
Failures in after hooks are not represented at the Feature Overview level.

### ISSUE THREAD

**Issue by SlyAtBest** [damianszczepanik/cucumber-reporting#310]
Hi,

we have had instances where a build in Jenkins has been marked as unstable due to test failures.  However when we look at the feature overview page, both the Scenario and Steps graphs show a 100% pass rate and each feature is marked as "passed".

The only way to determine the cause of the instability rating is to check the console output of Jenkins to ascertain which feature has failed.  Then you can drill down into that feature in the report to see the hook failure information and screenshots etc.

Whilst technically the scenario and it's steps have passed, failures in the after hooks can represent a leak of state between scenarios.  In some instances this leakage can be then cause of subsequent test failures and this can be hard to trace as things currently stand.

Before hooks are not a problem as failures in those cause the scenario steps to be skipped and you already have an option for treating skipped steps as failures.  Perhaps failures in hooks should be treated as hard failures by default, or perhaps you could add an option to allow this behaviour?

**Comment by damianszczepanik** [comment]
Overview pages don't count hooks and backgrounds. This is because the same background can be used for more scenarios and then counting is ambiguous. This is intentional fix in #280 

If you have unstable/failure in background/hook phrase you'd probably have also the same result in scenario and then this will be counted properly.

If there is some exceptional case or my logic thinking about this is not correct let me know details and attach json file so I can verify my assumptions.

**Comment by tommywo** [comment]
Other plugin for jenkins - https://wiki.jenkins-ci.org/display/JENKINS/Cucumber+Test+Result+Plugin 
marks feature as failed when hook fails. Maybe a good idea is to not count background and hooks, but include them in setting status for feature.

**Comment by damianszczepanik** [comment]
Attach json. This idea sounds fine but I need to check with code

**Comment by SlyAtBest** [comment]
I think @tommywo has nailed down exactly what I was getting at.  I wouldn't expect the hooks or background steps to be counted as scenarios, but it would be great if they did effect the status of the feature.  We've also noticed that the plugin linked above handles the hooks differently.

We run a generic set of before and after hooks  for every scenario, so if a change to our software breaks one of the hooks we would want our attention drawn to it as fast as possible.  We run screens displaying the overview page so a high percentage of failures shown in the graphs would certainly achieve this. 
 
I've attached a subset of results which demonstrates the issue we had yesterday where an after hook started failing for each scenario, due to an increase in the amount of time a process was taking.

[testResult.zip](https://github.com/damianszczepanik/cucumber-reporting/files/98968/testResult.zip)
